nxpmicro-mfgtools: init at 1.3.191

bmilanov opened the pull request:
Add nxpmicro-mfgtools to nixpkgs

From the project's homepage [1]:

    Freescale/NXP I.MX Chip image deploy tools.

The project's only binary and library at the moment is UUU. NXP
might add more in the future, so making the package name:

    nxpmicro-mfgtools

instead of:

    nxp-uuu

or something similar.

Add a preConfigure step based on bbigras' suggestion.

[1]: https://github.com/NXPmicro/mfgtools

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bmilanov committed:
maintainers: add bmilanov

bbigras reviewed (changes requested):
I think it would be cleaner that way.

EDIT: derp, I didn't add the ~requested~ suggested changes.

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+}:
+
+stdenv.mkDerivation rec {
+  name    = "nxpmicro-mfgtools-1.3.191";
```
```suggestion
  pname = "nxpmicro-mfgtools";
  version = "1.3.191";
```

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+}:
+
+stdenv.mkDerivation rec {
+  name    = "nxpmicro-mfgtools-1.3.191";
+  version = "uuu_1.3.191";
```
```suggestion
  pname = "nxpmicro-mfgtools";
  version = "1.3.191";
```

bbigras replied:
According to https://nixos.org/nixpkgs/manual/#sec-using-stdenv , pname with set the name with the right version.

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+  src = fetchFromGitHub {
+    owner  = "NXPmicro";
+    repo   = "mfgtools";
+    rev    = version;
```
```suggestion
    rev    = "uuu_${version}";
```

bbigras commented:
You didn't check "Tested using sandboxing". Are you sure you are not using sandboxing? It's enabled by default in NixOS.

You can check by running: `nix-shell -p nix-info --run "nix-info -m"`

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+    homepage = https://github.com/NXPmicro/mfgtools;
+    license = licenses.bsd3;
+    maintainers = [ maintainers.bmilanov ];
+    platforms = [ "x86_64-linux" ];
```
Are you sure it's Linux only? The project's readme says:
> The real cross platform. Linux, Windows, MacOS(not test yet)

Maybe try `platforms.all`.

bmilanov replied:
It says it's cross-platform, but I wanted to add only the platform that I am testing on.

Should I add all stated working, or should I add only tested platforms?

bbigras replied:
I don't know what the NixOS policy exactly is on this but maybe it's fine if we put "all" and the ofborg bot doesn't complain while building for all the platforms it supports.

bmilanov replied:
13f3760 fingers crossed

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  preConfigure = "echo ${version} > .tarball-version";
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs       = [ bzip2 libusb1 libzip openssl pkg-config ];
```
```suggestion
  nativeBuildInputs = [ cmake pkg-config ];
  buildInputs       = [ bzip2 libusb1 libzip openssl ];
```

> build time only dependencies are declared in `nativeBuildInputs`

https://nixos.org/nixpkgs/manual/#reviewing-contributions-new-packages

I'm pretty sure `pkg-config` is only needed at build-time.

bbigras commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,38 @@
+  version = "uuu_1.3.191";
+
+  meta = with stdenv.lib; {
+    description = "uuu (Universal Update Utility), mfgtools 3.0";
```
> For brevity, don’t repeat the name of package — just describe what it does

Check https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

I think a description of what the program does would be better.

bmilanov replied:
I wonder if it's better to get the contents of the long description in this field. Then think about what to add in the long description itself.

bmilanov replied:
How about this: 5f5d2ee ?

bmilanov commented:
Apologies for the multiple commits, I just now saw that I can batch suggestions into one.

bmilanov commented:
> You didn't check "Tested using sandboxing". Are you sure you are not using sandboxing? It's enabled by default in NixOS.
> 
> You can check by running: `nix-shell -p nix-info --run "nix-info -m"`

Seems like I have:

    - sandbox: `yes`

bbigras commented:
@GrahamcOfBorg build zenith

bbigras commented:
Ops I pasted the wrong command.

bbigras commented:
@GrahamcOfBorg build nxpmicro-mfgtools

bbigras commented:
Result of `nixpkgs-review pr 91634` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- nxpmicro-mfgtools
</details>

bbigras commented:
I don't know if it matters for new packages but here's the closure size:

❯ nix path-info -S -h /nix/store/fkfi4ainw1iq49zp2jgpmmd0dyqrhixd-nxpmicro-mfgtools-1.3.191
/nix/store/fkfi4ainw1iq49zp2jgpmmd0dyqrhixd-nxpmicro-mfgtools-1.3.191	  47.5M

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/209

danieldk commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,45 @@
+  pname   = "nxpmicro-mfgtools";
+  version = "1.3.191";
+
+  meta = with stdenv.lib; {
+    description = "Freescale /NXP I.MX Chip image deploy tools";
+    longDescription = ''
+      UUU (Universal Update Utility) is a command line tool, evolved out of
+      MFGTools (aka MFGTools v3).
+
+      One of the main purposes is to upload images to I.MX SoC's using at least
+      their boot ROM.
+
+      With time, the need for an update utility portable to Linux and Windows
+      increased. UUU has the same usage on both Windows and Linux. It means the same
+      script works on both OS.
+    '';
+    homepage = https://github.com/NXPmicro/mfgtools;
+    license = licenses.bsd3;
+    maintainers = [ maintainers.bmilanov ];
+    platforms = platforms.all;
+  };
```
Canonical order: `meta` comes last in `mkDerivation`.

danieldk commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,45 @@
+      increased. UUU has the same usage on both Windows and Linux. It means the same
+      script works on both OS.
+    '';
+    homepage = https://github.com/NXPmicro/mfgtools;
```
Unquoted URLs are deprecated:

```suggestion
    homepage = "https://github.com/NXPmicro/mfgtools";
```

danieldk commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,45 @@
+  version = "1.3.191";
+
+  meta = with stdenv.lib; {
+    description = "Freescale /NXP I.MX Chip image deploy tools";
```
```suggestion
    description = "Freescale/NXP I.MX chip image deploy tools";
```

danieldk commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,45 @@
+
+  preConfigure = "echo ${version} > .tarball-version";
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs       = [ bzip2 libusb1 libzip openssl ];
```
`nativeBuildInputs` and `buildInputs` go between `src` and `preConfigure`.

danieldk commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5590,6 +5590,8 @@ in
 
   nwdiag = with python3Packages; toPythonApplication nwdiag;
 
+  nxpmicro-mfgtools = callPackage ../applications/science/electronics/nxpmicro-mfgtools { };
```
```suggestion
  nxpmicro-mfgtools = callPackage ../applications/science/electronics/nxpmicro-mfgtools { };
```

Many other derive flashing utilities (e.g. `esptool`, `stm32flash`) are in `development/tools/misc`, so that may be the more logical location.

danieldk commented on pkgs/applications/science/electronics/nxpmicro-mfgtools/default.nix:
```diff
@@ -0,0 +1,45 @@
+}:
+
+stdenv.mkDerivation rec {
+  pname   = "nxpmicro-mfgtools";
```
Maybe this should be just be called `mfgtools`? Not 100% sure though ;).

bmilanov replied:
Hi, thanks for all the feedback!

I am not sure which way's better.

Usually I would add the "nxpmicro-" as I don't know how many mfgtools can pop up in the future (doubt there will be any more), with the assumption that adding "namespaces" later would be harder. I am really tempted to make it only "mfgtools" as it's less text to write.

Your call. I can amend the commit based on what you decide.

danieldk replied:
Thanks for the changes. There was a small syntax error that caused evaluation to fail, so I fixed that. I also used `nixpkgs-fmt` to format the derivation.

I think the prefix may be better here. As you say, `mfgtools` is very generic. I don't think there will be another package with the same name, but with the prefix it's easier to recognize and quickly find.

danieldk replied:
Whoops, seems like we are force-pushing at the same time. If you only fixed the syntax error, I can push my changes again.

danieldk reviewed (changes requested):
Thank you for contributing to `nixpkgs`. I have added some more comments.

Could you squash the commits together? The first commit should add you to `maintainers.nix`, the second should add the derivation.

bmilanov committed:
nxpmicro-mfgtools: init at 1.3.191

Add nxpmicro-mfgtools to nixpkgs

From the project's homepage [1]:

    Freescale/NXP I.MX Chip image deploy tools.

The project's only binary and library at the moment is UUU. NXP
might add more in the future, so making the package name:

    nxpmicro-mfgtools

instead of:

    nxp-uuu

or something similar.

[1]: https://github.com/NXPmicro/mfgtools

bmilanov commented:
Just after I pushed I saw that you've changed the comma style because of the build error.

I will back off so you can force push to the proper commit.

danieldk reviewed (approved):
Thanks, looks great now!

Result of `nixpkgs-review pr 91634` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- nxpmicro-mfgtools
</details>

Tested binary to work.

bbigras commented:
Thanks @bmilanov for your contribution and perseverance.

Thanks @danieldk too.
